tokio-trace/src/subscriber.rs

```diff
@@ -17,3 +17,19 @@ where
 {
     ::dispatcher::with_default(&::Dispatch::new(subscriber), f)
 }
+
+/// Sets this subscriber as the global default for the duration of the entire program.
+/// Will be used as a fallback if no thread-local subscriber has been set in a thread (using `with_default`.)
+///
+/// Can only be set once; subsequent attempts to set the global default will fail.
+/// Returns whether the initialization was successful.
+///
+/// [span]: ../span/index.html
+/// [`Subscriber`]: ../subscriber/trait.Subscriber.html
+/// [`Event`]: ../event/struct.Event.html
+pub fn set_global_default<S>(subscriber: S) -> bool
```

Comment:
I think I'd prefer if this function returned a `Result<(), InitializationError>`, as `Result` is `#[must_use]`, which introduces higher misuse resistance (e.g., a users forgets to check if `set_global_default()` returns false). With `Result`, one could write `set_global_default(SomeSubscriber).expect("Unable to set default subscriber");` and crash immediately if there is some sort of logic failure (e.g., a global subscriber was set twice).